plugin/audit_log/audit_log.c

```diff
@@ -787,6 +802,42 @@ int is_event_class_allowed_by_policy(mysql_event_class_t class,
 
 
 static
+void audit_log_update_thd_local(MYSQL_THD thd,
+                                audit_log_thd_local *local,
+                                unsigned int event_class)
+{
+  DBUG_ASSERT(audit_log_include_accounts == NULL ||
+              audit_log_exclude_accounts == NULL);
+
+  if (event_class == MYSQL_AUDIT_CONNECTION_CLASS)
+  {
+    LEX_STRING priv_user, priv_host;
+    MYSQL_SECURITY_CONTEXT ctx;
+    my_svc_bool ret;
+
+    ret= thd_get_security_context(thd, &ctx);
+    DBUG_ASSERT(!ret);
```

Comment:
but can it really fail? `thd_get_security_context` is just a `thd->security_context()` and in most places in server it used like this:
```
...
thd->security_context()->host_or_ip().str,
thd->security_context()->priv_user().str,
...
```